### PR TITLE
Route Drive Activity user lookup through UserInfo

### PR DESCRIPTION
**Pull request by peterdrier**
## Summary
- remove the DriveActivity repository fallback that joined `AspNetUserLogins` and `Users`
- inject `IUserServiceRead` into `DriveActivityMonitorService` and lazily load a per-run Google provider-key -> `UserInfo` map only after Directory lookup fails
- use `UserInfo.Email` for unresolved Google `people/{id}` actors after the Directory lookup fails, while logging and continuing with the raw id if the UserInfo fallback is unavailable
- skip ambiguous duplicate Google provider keys rather than attributing an audit entry to the wrong user
- update the active service data-access map for the new boundary

## Verification
- `dotnet build Humans.slnx --disable-build-servers -v minimal`
- `dotnet test tests\Humans.Application.Tests\Humans.Application.Tests.csproj --no-build --disable-build-servers -v minimal --filter "FullyQualifiedName~DriveActivityMonitor"`
- `dotnet test tests\Humans.Application.Tests\Humans.Application.Tests.csproj --no-build --disable-build-servers -v minimal`
- `dotnet ef migrations has-pending-model-changes --project src\Humans.Infrastructure --startup-project src\Humans.Web --context HumansDbContext --no-build`
- `git diff --check`

## HUM0025 Count
- Current `origin/main`: 15
- This branch: 11 (`-4`)

### CODE REVIEW

File: src/Humans.Application/Services/GoogleIntegration/DriveActivityMonitorService.cs [hunk 1]
```diff
@@ -41,6 +43,7 @@ public async Task<int> CheckForAnomalousActivityAsync(CancellationToken cancella
 
         // Per-invocation cache for resolved people/ IDs to avoid repeated API calls.
         var peopleIdCache = new Dictionary<string, string>(StringComparer.Ordinal);
+        var googleUserInfoByProviderKey = await BuildGoogleUserInfoByProviderKeyAsync(cancellationToken);
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/GoogleIntegration/DriveActivityMonitorService.cs]: Fixed in `13dfe4593`: the UserInfo login index is now lazy, only loaded after Directory fails to resolve a `people/{id}`, and non-cancellation failures are logged once then fall back to the raw people id for the run. Added tests covering both no eager UserInfo load when Directory succeeds and continued monitoring when the UserInfo fallback throws.

**Comment by peterdrier** [comment]
## PR Review — 2026-05-27

**PR**: #813 — Route Drive Activity user lookup through UserInfo
**Branch**: `techdebt/2026-05-27-codex-8` → `main`
**Scope**: 7 files, 1 commit (HUM0025: 15 → 11, -4)

---

### Prior Review Status

- **Codex (P2) — Guard UserInfo fallback failures**: **FIXED** in `13dfe4593`. The eager pre-loop build is gone; map is loaded lazily via local function `GetGoogleUserInfoByProviderKeyAsync` only after `TryResolvePersonEmailAsync` returns null (svc.cs:383–389). Non-cancellation exceptions are caught, logged once, and set `googleUserInfoLookupUnavailable = true` so subsequent resources skip the fallback. `OperationCanceledException` is correctly rethrown. Tests cover both the no-eager-load-on-success path (`...ResolvesPeopleIdViaDirectoryConnector` asserts `_userService.DidNotReceive().GetAllUserInfosAsync(...)`) and the fallback-throws path (`LeavesRawPeopleIdWhenUserInfoFallbackFails`).

---

### Spec Compliance (vs. PR body summary; no linked issue)

1. **Remove repo fallback joining `AspNetUserLogins`/`Users`** — FULLY ADDRESSED. `TryResolveEmailByGoogleUserIdAsync` removed from interface + impl; matching `[Grandfathered]` markers stripped from both `DriveActivityMonitorRepository` and `UserRepository`.
2. **Inject `IUserServiceRead`, lazy per-run map** — FULLY ADDRESSED. Ctor takes `IUserServiceRead`; map starts null; only built after Directory miss; memoised for the run.
3. **`UserInfo.Email` for unresolved actors, log+continue with raw id on failure** — FULLY ADDRESSED.
4. **Skip ambiguous duplicate Google provider keys** — FULLY ADDRESSED. `LoadGoogleUserInfoByProviderKeyAsync` tracks an ambiguous HashSet, removes on second hit, logs warning. Covered by `SkipsAmbiguousUserInfoProviderKeys`.
5. **Update service data-access map** — FULLY ADDRESSED. Entry now correctly states cross-section reads via `IUserServiceRead.GetAllUserInfosAsync`.

**Architecture intent (HUM0025 reduction)**: Verified — cross-section reach into `AspNetUserLogins`/`Users` removed from `DriveActivityMonitorRepository`; replacement dependency is the section-read interface (`IUserServiceRead`), matching peters-hard-rules.md.

---

### Code Quality

**IMPORTANT — `UserInfo.Email` can be null; the ambiguous-key warning misfires for tombstoned accounts**
`src/Humans.Application/Services/GoogleIntegration/DriveActivityMonitorService.cs:405–434`. `UserInfo.Email` returns null when both `UserEmails` is empty and `IdentityEmailColumn` is null (GDPR-anonymized accounts). `LoadGoogleUserInfoByProviderKeyAsync` stores `userInfo` into the dictionary without checking for a usable email. If a null-email user collides on a Google `ProviderKey`, the "Skipping ambiguous Google provider key" warning fires for what is actually a tombstoned account. Observable behavior is safe (raw `people/{id}` is used downstream because `resolved` stays null at svc.cs:388), but the log message is misleading and there's no regression coverage. Fix: in `LoadGoogleUserInfoByProviderKeyAsync`, `continue` when `userInfo.Email is null` before insertion.

**IMPORTANT — Test coverage gap: no eager-load assertion across multiple distinct people-ids**
`tests/Humans.Application.Tests/GoogleIntegration/DriveActivityMonitorServiceTests.cs:134–162`. The existing "Directory succeeds" test only uses a single people-id. The core invariant the Codex fix asks for ("no eager UserInfo build when Directory resolves everything") would be more robustly covered by a test with two distinct `people/{id}` actors, both Directory-resolved, asserting `GetAllUserInfosAsync` received 0 calls.

**IMPORTANT — No test for null-email `UserInfo` skip path**
Same test file, `BuildUserInfo` helper always supplies a non-null `IdentityEmailColumn`. If the null-email skip in finding #1 is added, it should have a regression test (GDPR-anonymized account with a Google ExternalLogin → raw people-id fallback, no crash, no misleading ambiguous warning).

**MINOR — `EmptyGoogleUserInfoByProviderKey` sentinel naming**
`DriveActivityMonitorService.cs:26–27`. Intent ("fallback unavailable") is only carried by the field name. A short rename or a one-line comment would help the next reader.

---

### Bottom Line

The architectural intent (kill the cross-section repository fallback, route through `IUserServiceRead`) is cleanly executed and the prior Codex finding is genuinely fixed and tested. Three IMPORTANT items are all about one edge case — null `UserInfo.Email` from GDPR-anonymized accounts — and its missing test coverage. None block merge; worth a follow-up commit before promoting to upstream.

---
*Review by Claude Code `/pr-review`*

**Comment by peterdrier** [comment]
## Fixes Applied — 2026-05-27

**Requested**: three IMPORTANT findings from the prior review — null-email `UserInfo` skip, eager-load assertion across distinct people-ids, null-email regression test.

**Changes** (commit `f21e0a84d`):
- `DriveActivityMonitorService.LoadGoogleUserInfoByProviderKeyAsync`: added `if (userInfo.Email is null) continue;` after the `"Google"` provider check, before ambiguous-key tracking. GDPR-anonymized accounts are now silently excluded from the provider-key index, so they can't trigger a misleading "Skipping ambiguous Google provider key" warning or leave the resolved actor null without signal.
- `CheckForAnomalousActivityAsync_DoesNotLoadUserInfo_WhenAllPeopleIdsResolvedByDirectory`: two distinct `people/{id}` actors both resolved by Directory; asserts `GetAllUserInfosAsync` never called.
- `CheckForAnomalousActivityAsync_LeavesRawPeopleIdWhenUserInfoHasNullEmail`: `UserInfo` with `IdentityEmailColumn: null` and empty `UserEmails`; asserts audit entry uses raw `people/{id}` and no ambiguous-key warning is emitted.

**Re-review**:
- Finding 1 (null-email skip placement): **VERIFIED** — guard at the right point, doesn't regress dedup or happy path.
- Finding 2 (multi-id no-eager-load): **VERIFIED** — two distinct actors, `DidNotReceive` assertion present.
- Finding 3 (null-email fallback): **VERIFIED** — `UserInfo.Email` correctly resolves to null given the helper inputs; both assertions correct.

No new issues introduced. Minor sentinel-naming MINOR from the original review is left unaddressed by intent.

**Test status**: 3766 unit/domain/analyzer/web tests pass. 98 integration tests skipped — pre-existing Docker-unavailable infra on the review machine, unrelated to this PR.

---
*Fix & re-review by Claude Code `/pr-review`*
